### PR TITLE
111 completion check

### PR DESCRIPTION
**Pull request by SkyRiser568**
Task Tracker script will create subtasks and groups that track progress for all tasks that implement ITask interface. Individual tasks will need to update task tracker using the UpdateTaskTracker() public method.
Custom dictionary class is used to serialize dictionary entries into the inspector.

[x] I have run this branch locally
[x] I have performed a self-review of my code
[x] I have confirmed critical features still function

### CODE REVIEW

File: MonstieWash/Assets/Scripts/TaskTracker.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+public class TaskTracker : MonoBehaviour
+{
+    //Unity Inspector
+    //Container object for all the the different areas to clean 
+    [SerializeField] private Transform taskContainer;
+    //These are temporary replacement for UI progress bar
+    [SerializeField] private SerializableDictionary<string, float> m_taskProgress = new SerializableDictionary<string, float>();
+
+    //Private
+    private List<string> m_taskDictKeys = new List<string>();
```
Comment (CrispyKorn) [inline MonstieWash/Assets/Scripts/TaskTracker.cs:16]: Same with here, try to use new().
```suggestion
    private List<string> m_taskDictKeys = new();
```

**Review by CrispyKorn — CHANGES_REQUESTED** [review]
Wow, what a behemoth of a feature, really great work figuring this one out! There's some changes to be made but they should be pretty easy fixes. We can figure out some of the weirder stuff another time because it doesn't affect the functionality of the feature.

**Comment by SkyRiser568** [comment]
Thanks for looking over this, your feedback is very helpful! I definitely let some bad code slip through while coding this, and will try to be more aware in the future. I'll have the changes done either tonight or tomorrow.

**Review by CrispyKorn — APPROVED** [review]
Awesome, nice work! Looks good to me now 👍

**Pull request merged**